content/backend/typescript-apollo/2-a-simple-query.md

```diff
@@ -0,0 +1,329 @@
+
+<Instruction>
+
+Import `src/graphql` in `src/schema.ts`:
+
+```typescript{3,6}(path="../hackernews-typescript/src/schema.ts")
+import { makeSchema } from "nexus";
+import { join } from "path";
+import * as types from "./graphql";   // 1
+
+export const schema = makeSchema({
+    types,   // 2
+    outputs: {
+        typegen: join(__dirname, "..", "nexus-typegen.ts"), 
+        schema: join(__dirname, "..", "schema.graphql"), 
+    },
+});
+```
+
+</Instruction>
+
+Let's go through the changes marked with comments:
+
+1. You are importing the `graphql` model which exports the `Link` objectType through `index.ts`. The import is named `types`. 
+
+2. You are passing `types` to the `makeSchema` function. Nexus will do its thing to generate the SDL from this. 
+
+Now let's take a look at what has changed in `schema.graphql`: 
+
+```graphql{1-5}(path=".../hackernews-typescript/schema.graphql"&nocopy)
+type Link {
+  description: String!
+  id: Int!
+  url: String!
+}
+
+type Query {
+  ok: Boolean!
+}
+
+```
+
+
+> **Note**: If you have a terminal running with `npm run dev`, your schema should get updated automatically. Otherwise, you can always run `npm run generate` to make nexus regenerate your schema. 
+
+If you take a look at `nexus-typegen.ts`, you will see that it has a new `Link` interface as well. This is really convenient because the interface signature is identical to that of the `link` type in your GraphQL schema. 
+
+```typescript{2-6}(path=".../hackernews-typescript/nexus-typegen.ts"&nocopy)
+export interface NexusGenObjects {
+  Link: { // root type
+    description: string; // String!
+    id: number; // Int!
+    url: string; // String!
+  }
+  Query: {};
+}
+```
+
+> **Note**: One of the major advantages of the code-first approach is that you don't have to worry about having your GraphQL types and your TypeScript types going out of sync. Since nexus is the source of truth that generates both, there is no risk of the two mismatching. 
+
+### Implementing the feed query
+
+Now that the `Link` type is ready, you will create a `feed` query to return all the created `link` objects. 
+
+<Instruction>
+
+Make the following changes to implement the `feed` query in `src/graphql/Link.ts`:
+
+```typescript{1-2,13-37}(path="../hackernews-typescript/src/graphql/Link.ts")
+import { extendType, objectType } from "nexus";
+import { NexusGenObjects } from "../../nexus-typegen";  
+
+export const Link = objectType({
+    name: "Link", // <- Name of your type
+    definition(t) {
+        t.nonNull.int("id"); 
+        t.nonNull.string("description"); 
+        t.nonNull.string("url"); 
+    },
+});
+
+let links : NexusGenObjects["Link"][]= [   // 1
+    {
+        id: 1,
+        url: "www.howtographql.com",
+        description: "Fullstack tutorial for GraphQL",
+    },
+    {
+        id: 2,
+        url: "graphql.org",
+        description: "GraphQL official website",
+    },
+];
+
+export const PostQuery = extendType({  // 2
```

Comment:
I noticed that this is called `LinkQuery` later, is the name intentionally different here?

Reply:
Unintentional. I refactored it to `LinkQuery` which in my opinion seems like a more apt name, fixing!